whd/dispatcher.py

```diff
@@ -211,6 +213,106 @@ def _set_pr_labels(self, pr_event, resources):
                     f" in '{repository_path}' found. Ignoring, since they are not an org member'."
                 )
 
+    def _start_not_triggered_jobs(self, concourse_api, resources):
```

Comment:
please rename this method, as already remarked before. this is a very confusing name (for me).

how about: `ensure_pull_requests_jobs_are_run` or sth similar?

also, this method is nested way too deeply, and does too many things. the filtering for PR-jobs e.g. could be done before.

ideally, there should be:
- one method that accepts a PR, and determines all related PR-resource versions + jobs triggered by it
- one method that accepts a PR-resource version and a job, that will ensure said job is executed exactly once for the given PR resource version


as a general note, please refrain from using `filter` / `map` built-ins, and rather use dict, set, list, generator comprehensions (depending on what is needed), unless there is a good reason not to